pkg/controller/ingress-shim/controller.go

```diff
@@ -161,7 +128,53 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return err
 	}
 
-	return c.Sync(ctx, crt)
+	return c.sync(ctx, crt)
+}
+
+// Whenever a Certificate gets deleted, we want to reconcile its parent Ingress.
+// This parent Ingress is called "controller object". For example, the following
+// Certificate is controlled by the Ingress "example":
+//
+//     kind: Certificate
+//     metadata:
+//       namespace: cert-that-was-deleted
+//       ownerReferences:
+//       - controller: true                                       ← this
+//         apiVersion: networking.k8s.io/v1beta1
+//         kind: Ingress
+//         name: example
+//         blockOwnerDeletion: true
+//         uid: 7d3897c2-ce27-4144-883a-e1b5f89bd65a
+func certificateDeleted(queue workqueue.RateLimitingInterface) func(obj interface{}) {
+	return func(obj interface{}) {
+		cert, ok := obj.(*cmapi.Certificate)
+		if !ok {
+			runtime.HandleError(fmt.Errorf("not a Certificate object: %#v", obj))
+			return
+		}
+
+		ingress := metav1.GetControllerOf(cert)
+		if ingress == nil {
+			// No controller should care about orphans being deleted or
+			// updated.
+			return
+		}
+
+		// We don't check the apiVersion e.g. "networking.k8s.io/v1beta1"
+		// because there is no chance that another object called "Ingress" be
+		// the controller of a Certificate.
+		if ingress.Kind != "Ingress" {
+			return
+		}
+
+		// Owner references don't know about the namespace of the referenced
+		// object. That's because owner refs do not support cross-namespace
+		// references. We also know that the Certificate and its parent Ingress
+		// must both be on the same namespace. We thus use the Certificate's
+		// namespace to trigger a resync of the parent Ingress (the string below
+		// is a "key" of the form "namespace-1/my-ingress").
```

Comment:
I think in this case the comment isn't really necessary as this is just how it works. That's
just part of Kubernetes, so if someone finds it confusing they will look at the ref and 
see that it doesn't have the namespace. I wouldn't want to duplicate this comment
every time a ref was used.

Reply:
Good point, it seems like this comment is paraphrasing how Kubernetes works. That said, `ownerReference` not having the object's namespace is something not many people are aware of (typical Kubernetes tribal knowledge). 

I removed the comment 👍